src/protocols/kubernetes/url.c

```diff
@@ -109,26 +110,52 @@ int guac_kubernetes_endpoint_attach(char* buffer, int length,
                 sizeof(escaped_pod), kubernetes_pod))
         return 1;
 
-    /* Generate attachment endpoint URL */
-    if (kubernetes_container != NULL) {
+    /* Generate endpoint path depending on the call type */
+    char* call = "attach";
+    if (exec_command != NULL)
+        call = "exec";
+
+    char endpoint_path[GUAC_KUBERNETES_MAX_ENDPOINT_LENGTH];
+
+    written = snprintf(endpoint_path, sizeof(endpoint_path), 
+        "/api/v1/namespaces/%s/pods/%s/%s", escaped_namespace, escaped_pod, call);
+        
+    if (written < 0 || written >= sizeof(endpoint_path))
+        return 1;
+
+    /* Generate endpoint params */
+    char endpoint_params[GUAC_KUBERNETES_MAX_ENDPOINT_LENGTH]="";
+
+    if (exec_command != NULL) {
+        /* Escape exec command */
+        if (guac_kubernetes_escape_url_component(escaped_exec_command,
+                    sizeof(escaped_exec_command), exec_command))
+            return 1;
 
+        written = snprintf(endpoint_params, sizeof(endpoint_params), 
+            "command=%s&", escaped_exec_command);
+
+        if (written < 0 || written >= sizeof(endpoint_params))
+            return 1;
+    }
```

Comment:
Perhaps there is a way to clean up these repeated, similar calls to `snprintf()` intended to append parameters to a URI? Seems the sort of thing that would benefit greatly from being moved into its own simple, testable function.

Reply:
Fixed